aptos-move/framework/aptos-framework/sources/dora_committee.move

```diff
@@ -0,0 +1,1042 @@
+        rpc_port: vector<u16>
+    ) acquires CommitteeInfoStore, SupraCommitteeEventHandler {
+        // Assert the length of the vector for two are the same
+        assert!(
+            vector::length(&ids) == vector::length(&node_addresses),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            vector::length(&ids) == vector::length(&ip_public_address),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            vector::length(&ids) == vector::length(&dora_public_key),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            vector::length(&ids) == vector::length(&network_public_key),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            vector::length(&ids) == vector::length(&elgamal_pub_key),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            vector::length(&ids) == vector::length(&network_port),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        assert!(
+            vector::length(&ids) == vector::length(&rpc_port),
+            error::invalid_argument(INVALID_COMMITTEE_NUMBERS)
+        );
+        while (vector::length(&ids) > 0) {
+            let id = vector::pop_back(&mut ids);
+            let node_address = vector::pop_back(&mut node_addresses);
+            let ip_public_address = vector::pop_back(&mut ip_public_address);
+            let dora_public_key = vector::pop_back(&mut dora_public_key);
+            let network_public_key = vector::pop_back(&mut network_public_key);
+            let elgamal_pub_key = vector::pop_back(&mut elgamal_pub_key);
+            let network_port = vector::pop_back(&mut network_port);
+            let rpc_port = vector::pop_back(&mut rpc_port);
+            upsert_committee_member(
+                com_store_addr,
+                owner_signer,
+                id,
+                node_address,
+                ip_public_address,
+                dora_public_key,
+                network_public_key,
+                elgamal_pub_key,
+                network_port,
+                rpc_port
+            );
+        }
+    }
+
+    /// Remove the node from the committee
+    public entry fun remove_committee_member(
+        com_store_addr: address,
+        owner_signer: &signer,
+        id: u64,
+        node_address: address
+    ) acquires CommitteeInfoStore, SupraCommitteeEventHandler {
+        // Only the OwnerCap capability can access it
+        let _acquire = &capability::acquire(owner_signer, &OwnerCap {});
+
+        let committee_store = borrow_global_mut<CommitteeInfoStore>(com_store_addr);
+        let committee = simple_map::borrow_mut(&mut committee_store.committee_map, &id);
+        assert!(simple_map::contains_key(&committee.map, &node_address), error::invalid_argument(NODE_NOT_FOUND));
```

Comment:
Can we create common internal functions like `ensure_node_address_exist()`, and use it everywhere
```
fun ensure_node_address_exist() {
     assert!(is_node_address_exist(&committee.map, &node_address), error::invalid_argument(NODE_NOT_FOUND))
}
```

Reply:
@nizam-supraoracles Thanks, addressed in new commit.